akirachix/Magezi-Frontend
# added sidebar

# Engineers Checklist

## What this PR does?
This PR introduces a sidebar component for navigation within the application. The sidebar aims to improve user navigation by providing quick access to essential features like Dashboard, Contracts, Payments, and Users. This enhances user experience and streamlines interactions within the application.

**Dependencies:**

react-icons: Required for iconography used in the sidebar.

* What is this?
This is a new sidebar component that provides navigation links for key areas of the application.

* Why this?
To enhance user experience by providing a clear and accessible navigation structure.

* How are you doing this? 
Added a sidebar component for navigation.
Integrated icons from react-icons.
Implemented dynamic styling based on the open/closed state.
Added a settings section with dropdown functionality.

## Type of change

Please delete options that are not relevant.


## How can it be tested?
Pull the latest changes from the repository.
Run the application using npm run dev.
Navigate to the main application page and verify the sidebar appears.
Test the sidebar links to ensure they navigate to the correct pages.
Toggle the sidebar open and closed to confirm the functionality works as expected.
Check the settings section to ensure the dropdown functionality operates correctly.

# Where is it documented?
[https://nakatozainamusana.atlassian.net/jira/software/projects/SCRUM/boards/4/backlog?selectedIssue=SCRUM-52](url)


![image](https://github.com/user-attachments/assets/c7512225-c74d-46f0-9543-5f8b8f447fc4)
